### PR TITLE
upgrade to postgres 12 with jit off

### PR DESCRIPTION
**Pull request by denis-yuen**
companion for https://github.com/dockstore/dockstore/pull/4388 

comparable performance to postgres 11 if we turn off jit 
background at https://severalnines.com/database-blog/overview-just-time-compilation-jit-postgresql or hhttps://dba.stackexchange.com/questions/257759/recursive-cte-based-on-function-values-significantly-slower-on-postgres-12-than/258493#258493
guessing this might be a testing only issue since llvm should be available in the docker image, but we are inserting and dropping data rapidly which may be throwing off the query planner

### CODE REVIEW

**Comment by denis-yuen** [comment]
```
[INFO] Tests run: 13, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 846.903 s - in io.dockstore.webservice.WebhookIT
[INFO] testGithubDirDockstoreYml(io.dockstore.webservice.WebhookIT)  Time elapsed: 26.859 s
[INFO] testTestParameterPaths(io.dockstore.webservice.WebhookIT)  Time elapsed: 24.65 s
[INFO] testDescriptorChange(io.dockstore.webservice.WebhookIT)  Time elapsed: 61.915 s
[INFO] testTools(io.dockstore.webservice.WebhookIT)  Time elapsed: 69.005 s
[INFO] testGitHubReleaseNoWorkflowOnDockstore(io.dockstore.webservice.WebhookIT)  Time elapsed: 128.18 s
[INFO] testWorkflowMigration(io.dockstore.webservice.WebhookIT)  Time elapsed: 89.227 s
[INFO] testGitHubReleaseNoWorkflowOnDockstoreNoUser(io.dockstore.webservice.WebhookIT)  Time elapsed: 25.132 s
[INFO] testDockstoreYmlFilters(io.dockstore.webservice.WebhookIT)  Time elapsed: 129.888 s
[INFO] testDockstoreYmlAuthors(io.dockstore.webservice.WebhookIT)  Time elapsed: 86.305 s
[INFO] testInvalidDockstoreYmlFiles(io.dockstore.webservice.WebhookIT)  Time elapsed: 48.153 s
[INFO] testManualRefreshWorkflowWithGitHubApp(io.dockstore.webservice.WebhookIT)  Time elapsed: 30.555 s
[INFO] testDockstoreYmlPublish(io.dockstore.webservice.WebhookIT)  Time elapsed: 36.618 s
[INFO] testDeleteDefaultWorkflowVersion(io.dockstore.webservice.WebhookIT)  Time elapsed: 78.215 s
``` 
No good

**Comment by denis-yuen** [comment]
```
[INFO] Tests run: 13, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 885.169 s - in io.dockstore.webservice.WebhookIT
[INFO] testGithubDirDockstoreYml(io.dockstore.webservice.WebhookIT)  Time elapsed: 28.049 s
[INFO] testTestParameterPaths(io.dockstore.webservice.WebhookIT)  Time elapsed: 25.924 s
[INFO] testDescriptorChange(io.dockstore.webservice.WebhookIT)  Time elapsed: 63.284 s
[INFO] testTools(io.dockstore.webservice.WebhookIT)  Time elapsed: 69.913 s
[INFO] testGitHubReleaseNoWorkflowOnDockstore(io.dockstore.webservice.WebhookIT)  Time elapsed: 171.052 s
[INFO] testWorkflowMigration(io.dockstore.webservice.WebhookIT)  Time elapsed: 109.252 s
[INFO] testGitHubReleaseNoWorkflowOnDockstoreNoUser(io.dockstore.webservice.WebhookIT)  Time elapsed: 26.693 s
[INFO] testDockstoreYmlFilters(io.dockstore.webservice.WebhookIT)  Time elapsed: 123.056 s
[INFO] testDockstoreYmlAuthors(io.dockstore.webservice.WebhookIT)  Time elapsed: 77.394 s
[INFO] testInvalidDockstoreYmlFiles(io.dockstore.webservice.WebhookIT)  Time elapsed: 43.372 s
[INFO] testManualRefreshWorkflowWithGitHubApp(io.dockstore.webservice.WebhookIT)  Time elapsed: 27.933 s
[INFO] testDockstoreYmlPublish(io.dockstore.webservice.WebhookIT)  Time elapsed: 34.45 s
[INFO] testDeleteDefaultWorkflowVersion(io.dockstore.webservice.WebhookIT)  Time elapsed: 72.553 s
```
nope, 12.0 is slow too

**Comment by denis-yuen** [comment]
Notes on factors to rule out:
- update to dropwizard 4.0.24 and associated updates to hibernate 5.5.3 and liquibase 3.10.3
-

File: docker-compose.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@ services:
     # networks:
       # - elastic
   postgres_db:
-    image: postgres:11.12
+    image: postgres:12.7
```
Comment (coverbeck) [inline docker-compose.yml:45]: Add command here as well?

**Review by coverbeck — COMMENTED** [review]
Are you closing this PR in favor of the 13.x one?

**Comment by denis-yuen** [comment]
> Are you closing this PR in favor of the 13.x one?

No, the performance is slightly different and just in case of other issues we discover, I think I'll squash this one before the postgres 13 one so we can pick and choose whether to revert to 12 or 11 "next" time. 

> Add command here as well?

Undecided, I think I'll create a ticket to investigate when we get the performance testing back up. We're second guessing the query planner here and while that pays off in our tests, I'm not sure that is the case for "real".

**Review by garyluu — APPROVED** [review]

**Pull request merged**